espnet2/enh/decoder/codec_decoder.py

```diff
@@ -0,0 +1,89 @@
+import torch
+import torch.nn.functional as F
+import torchaudio.transforms as T
+
+from espnet2.enh.decoder.abs_decoder import AbsDecoder
+from espnet2.speechlm.tokenizer.codec_tokenizer import CodecTokenizer
+
+
+class CodecDecoder(AbsDecoder):
+    """Codec decoder for speech enhancement and separation"""
+
+    def __init__(
+        self,
+        channel: int,  # this is the expected output size of the continuous codec output
+        codec_choice: str,
+        codec_fs: int,
+        sample_fs: int,
+        device: str = "cpu",
+        dump_audio: bool = False,
+        checkpoint_path: str = None,
+        config_path: str = None,
+        max_token_per_frame: int = 32,
+    ):
```

Comment:
It is better to add a docstring to explain each argument here. For example, what is the difference between `codec_fs` and `sample_fs`, and how can we prepare a customized `checkpoint_path` and `config_path` (what should these files look like)?